### PR TITLE
1.0

### PR DESCRIPTION
**Pull request by dr-frmr**
proposed 1.0 changes: add `our()`, change `spawn`'s `grant-capabilities` argument to be more general, adjust comments

### CODE REVIEW

File: kinode.wit [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+    /*    System Types    */
+    /*      ˗ˏˋ ♡ ˎˊ˗     */
 
     // JSON is passed over WASM boundary as a string.
     type json = string;
 
+    // In types passed from kernel, node-id will be a valid Kimap entry.
     type node-id = string;
 
     // Context, like a message body, is a protocol-defined serialized byte
```
Comment (nick1udwig) [inline kinode.wit]: Should we change these `//` comments throughout to `///` comments? They will then be docs comments, similar to Rust https://component-model.bytecodealliance.org/design/wit.html#documentation

File: kinode.wit [hunk 2]
```diff
@@ -1,13 +1,15 @@
-package kinode:process@0.8.0;
+package kinode:process@1.0.0;
 
 interface standard {
-    //
-    // System types:
-    //
+
+    /*      ˗ˏˋ ♡ ˎˊ˗     */
+    /*    System Types    */
```
Comment (nick1udwig) [inline kinode.wit]: Given comment below (change `//` -> `///`), I'd prefer these `/*`s be `//`s. I think you changed them in order to have a different section header from the docs comments, but the `//` vs `///` will differentiate and have the benefit of not looking bad on non-monospaced viewers (like Github; check the diff here or View file)

File: kinode.wit [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+    /*      ˗ˏˋ ♡ ˎˊ˗     */
 
     // JSON is passed over WASM boundary as a string.
```
Comment (nick1udwig) [inline kinode.wit]: `WASM` -> `Wasm`

File: kinode.wit (hunk 2 above)
Comment (nick1udwig) [reply kinode.wit]: image

File: kinode.wit [hunk 4]
```diff
@@ -106,40 +108,49 @@ interface standard {
...
+    set-on-exit: func(on-exit: on-exit);
+
     get-state: func() -> option<list<u8>>;
 
     set-state: func(bytes: list<u8>);
 
     clear-state: func();
 
     spawn: func(
+        // name is optional. if not provided, name will be a random u64.
```
Comment (nick1udwig) [inline kinode.wit:141]: Should these "in-line" comments also follow the capitalization convention we are using in the "top-level" comments?

File: kinode.wit [hunk 5]
```diff
@@ -85,10 +87,10 @@ interface standard {
 
     // Network errors come from trying to send a message to another node.
```
Comment (nick1udwig) [inline kinode.wit]: Is it true that these can only happen when sending to another node? E.g. consider the case where I send a message to a local process and it times out. Does that not give back a SendError::Timeout?

If so, I recommend we change or remove this introductory sentence: I would assume, reading this first sentence, that it applied to this type in general (i.e. that a SendError is the same thing as a network error).

**Review by nick1udwig — APPROVED** [review]
All comment-related.

On that front: on line 65 there's a sentence that starts with a lower case:
```
A message can be a request or a response. within a response, there is
```
should be
```
A message can be a request or a response. Within a response, there is
```

File: kinode.wit (hunk 2 above)
Comment (dr-frmr) [reply kinode.wit]: Oh good catch. Will swap in `//`

File: kinode.wit (hunk 4 above)
Comment (dr-frmr) [reply kinode.wit:141]: I differentiated on purpose

**Pull request merged**